encled

```diff
@@ -84,6 +90,62 @@ def find_name(enc, slot):
         name = '0' + name  # try same name with more zeroes at front
     return None
 
+def find_disk( a_disk_name ):
+    '''
+        Explore /dev/sd* and /dev/disk/* to find hard disk device
+        If a hard disk is found then
+            return a string like "/dev/sdXX"
+        else
+            return None
+    '''
+    # remove /dev prefix if any
+    a_disk_name = os.path.basename( a_disk_name )
+
+    # create subdirectory list from /dev/disk/*
+    disk_directories = [ os.path.join("/dev/disk/",d) for d in os.listdir("/dev/disk/") ]
+    disk_directories.append( "/dev/" )
+
+    for subdir_path in disk_directories:
+        files = os.listdir( subdir_path )
+        for f in files:
+            if re.search(a_disk_name, f, re.IGNORECASE): # the file match
```

Comment:
Do you want to support regexps? How do you plan to use it? `sdled s.+z`?

May be we can just check for case-insensitive match?